django/db/backends/base/schema.py

```diff
@@ -804,6 +804,9 @@ def alter_field(self, model, old_field, new_field, strict=False):
         If `strict` is True, raise errors if the old column does not match
         `old_field` precisely.
         """
+        if self._is_moving_model_field(old_field, new_field):
+            return
```

Comment:
We cannot assume that it's a no-op as other changes can be made at the same time. For example, I moved `test_one.MyModel` to `test_two.MyModel` and added `null=True` one of `ForeignKey`:
```bash
$ python manage.py makemigrationsWas the model test_one.MyModel moved to test_two.MyModel? [y/N] y
Migrations for 'test_one':
  test_one/migrations/0002_alter_mymodel_options.py
    - Change Meta options on mymodel
  test_one/migrations/0003_delete_mymodel_alter_myrelatedmodels_field_fk_and_more.py
    - Delete model mymodel
    - Alter field field_fk on myrelatedmodels
    - Alter field field_m2m on myrelatedmodels
Migrations for 'test_two':
  test_two/migrations/0002_mymodel_alter_myothermodel_field_fk.py
    - Create model MyModel
    - Alter field field_fk on myothermodel

$ python manage.py sqlmigrate test_two 0002
BEGIN;
--
-- Create model MyModel
--
-- (no-op)
--
-- Alter field field_fk on myothermodel
--
-- (no-op)
COMMIT;
```

Reply:
Looking into this 👍

Reply:
pushed changes that handle base cases, now working on edge cases of AlterField like changing non db attributes, ignored attributes etc.

Reply:
> now working on edge cases of AlterField like changing non db attributes, ignored attributes etc.

This is already handled by `if not self._field_should_be_altered(old_field, new_field)`. Isn't it enough to swap these checks?

Reply:
The `to` kwarg was being a barrier in swapping checks. Since the value of both `to` were different in `old_kwargs` and `new_kwargs`, it was returning `True`.